sample_keyprovider/README.md

```diff
@@ -1,20 +1,61 @@
-# README
+# CoCo Keyprovider
 
-Sample Keyprovider is a very simple Keyprovider skeleton, which currently has the following functions:
+CoCo Keyprovider is a very simple keyprovider tool, which can help to generate CoCo-compatible encrypted images.
+The encrypted image can be decrypted using `cc-kbc`, `offline-fs-kbc`, `offline-sev-kbc`, `online-sev-kbc` and `eaa-kbc`.
 
-1. It can be used to encrypt the container image to test the KBC module in AA. Sample Keyprovider provides a modular encryption mechanism. Different encryption modules can be used according to different kbcs to be tested.
+Also, historical toy KBC `sample kbc` is also supported.
 
-Sample Keyprovider can be a reference code base for you to quickly build your own KBS.
+The following guide will help make an encrypted image using CoCo keyprovider and decrypt that.
 
-## Test KBC modules
+## Related tools
 
-### Encryption
+Let's firstly introduce the related tools to help generate keys.
 
-Build and run sample keyprovider. You must explicitly specify the encryption module to use through the --features parameter (sample_enc for example): 
+### Test Key Generation and Offline-Fs-KBC
```

Comment:
This chapter is specific to Offline-Fs-KBC, so it is recommended to put it at the end and describe how to test Offline-Fs-KBC in a separate chapter.

The main body of README should guide the user to directly generate an independent key.

Reply:
Yes, reordered